src/test/java/ibm/jceplus/junit/base/BaseTestECDSASignature.java

```diff
@@ -759,6 +768,19 @@ public void testPostiveSigBytes() throws Exception {
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"secp256r1", "secp384r1", "secp521r1"})
+    public void testECDSASignatureWithInvalidKeySpec(String curveName) throws Exception {
+        ECPrivateKey ecPrivKey = generateInvalidPrivateKey(curveName);
+        Signature sig = Signature.getInstance("SHA256withECDSA", this.getProviderName());
+        try {
+            sig.initSign(ecPrivKey);
+            fail("Expected <java.security.InvalidKeyException> to be thrown");
+        } catch (java.security.InvalidKeyException ike) {
+            System.out.println("Expected exception <java.security.InvalidKeyException> for ECDSA/SHA256withECDSA/" + curveName + "is caught.");
```

Comment:
Should we split this line into multiple?

Reply:
splited